test/parallel/test-net-write-after-close.js

```diff
@@ -21,24 +21,27 @@
 
 'use strict';
 const common = require('../common');
+
 const net = require('net');
 
+let serverSocket;
+
 const server = net.createServer(common.mustCall(function(socket) {
+  serverSocket = socket;
+
   socket.resume();
 
   socket.on('error', common.mustCall(function(error) {
     console.error('got error, closing server', error);
     server.close();
   }));
-
-  setTimeout(common.mustCall(function() {
-    console.error('about to try to write');
-    socket.write('test', common.mustCall());
-  }), 250);
 }));
 
 server.listen(0, function() {
   const client = net.connect(this.address().port, function() {
+    client.on('end', () => {
```

Comment:
Test fails if I replace `end` with `finish` because `serverSocket` is `undefined`...

```console
/Users/trott/io.js/test/parallel/test-net-write-after-close.js:43
      serverSocket.write('test', common.mustCall());
                  ^

TypeError: Cannot read property 'write' of undefined
    at Socket.client.on (/Users/trott/io.js/test/parallel/test-net-write-after-close.js:43:19)
    at emitNone (events.js:110:20)
    at Socket.emit (events.js:207:7)
    at finishMaybe (_stream_writable.js:587:14)
    at endWritable (_stream_writable.js:595:3)
    at Socket.Writable.end (_stream_writable.js:546:5)
    at Socket.end (net.js:491:31)
    at Socket.<anonymous> (/Users/trott/io.js/test/parallel/test-net-write-after-close.js:45:12)
    at Object.onceWrapper (events.js:314:30)
    at emitNone (events.js:105:13)
```